### DISCUSSION TITLE
Code Contributions to Microsoft.SqlServer.DacFx

### DISCUSSION THREAD

**Discussion by abbgrade** [microsoft/DacFx#369] [General]
Hello,

are there any plans to move the source code of Microsoft.SqlServer.DacFx to this or another public repo and to allow code contributions?
I ask, since I'd like to try to implement my feature request #226 on myself.

In case you don't want to publish it, since the code quality is embarrassing, please publish it anyway and blame someone else ;)

**Comment by dzsquared** [comment]
@abbgrade - although new development is generally done in the open, we've also been able to slowly move through open sourcing different libraries for SQL tooling. As you noted- DacFx isn't one of them.

- started in the open - [SqlToolsService](https://github.com/microsoft/sqltoolsservice)
- open sourced - [SqlManagementObjects](https://github.com/microsoft/sqlmanagementobjects), [ScriptDOM](https://github.com/microsoft/sqlscriptdom)

There's interest in moving DacFx into open source, but there's no timeline for it at this point.

Nice work on PsDac! I'm glad you've been able to take DacFx further to solve and streamline DB interactions.
